packages/core/content-releases/server/src/controllers/release-action.ts

```diff
@@ -17,6 +18,45 @@ const releaseActionController = {
       data: releaseAction,
     };
   },
+  async findMany(ctx: Koa.Context) {
+    const releaseId: GetReleaseActions.Request['params']['releaseId'] = ctx.params.releaseId;
+    const allowedContentTypes = getAllowedContentTypes({ strapi, userAbility: ctx.state.userAbility });
+
+    // We create an object with the permissionsChecker for each contentType, then we can reuse it for sanitization
+    const permissionsChecker: Record<UID.ContentType, any> = {};
+    // We create a populate object for polymorphic relations, so we considered custom conditions on permissions
+    const morphSanitizedPopulate: Record<UID.ContentType, any> = {};
+
+    for (const contentTypeUid of allowedContentTypes) {
+      const permissionChecker = await getPermissionsChecker({ strapi, userAbility: ctx.state.userAbility, model: contentTypeUid });
+      permissionsChecker[contentTypeUid] = permissionChecker; 
+      morphSanitizedPopulate[contentTypeUid] = await permissionChecker.sanitizedQuery.read({});
+    }
+
+    const releaseService = getService('release', { strapi });
+    const { results, pagination } = await releaseService.findActions(releaseId, allowedContentTypes, { 
+      populate: {
+        entry: {
+          on: morphSanitizedPopulate
+        }
+      }
```

Comment:
as you pointed out, this doesn't filter the actions with entries the user can not read, is this enough for the use case?

Reply:
It's okay for now, in this way we return the release action with the entry field = null. So we make sure we are not returning anything not allowed.

We are moving forward with this approach and probably after the alpha we are going to customize this to use knex  and handle the use case of custom conditions

Reply:
sounds good to me :)